cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -74,9 +64,10 @@ private class DefOrUse extends TDefOrUse {
   /** Gets the block of this definition or use. */
   abstract IRBlock getBlock();
 
-  /** Holds if this definition or use has rank `rank` in block `block`. */
-  cached
-  final predicate hasRankInBlock(IRBlock block, int rnk) { rnk = getRank(this, block) }
+  /** Holds if this definition or use has index `index` in block `block`. */
+  final predicate hasIndexInBlock(IRBlock block, int index) {
```

Comment:
Previously, `hasRankInBlock` could only hold for `rnk \in [1, ...]` because it delegates to `getRank` which uses a `rank` aggregate (and rank aggregates start at 1).
But `hasIndexInBlock` will now also hold for `index in [0, ...]` because it delegates to `block.getInstruction` which is zero-based I assume.

Isn't this a behavior change or is `rnk`/`index` only used transparently and so this does not matter?

Reply:
Good point. This is _completely_ behavior preserving, but all current uses of this predicate shouldn't depend on this change for the exact reason that you mentioned :)